packages/react-instantsearch-hooks-web/src/widgets/SearchBox.tsx

```diff
@@ -40,6 +59,14 @@ export function SearchBox({ queryHook, ...props }: SearchBoxProps) {
     setQuery(event.currentTarget.value);
   }
 
+  function onSubmit(event: React.FormEvent<HTMLFormElement>) {
+    if (props.onSubmit) {
+      props.onSubmit(event);
+    } else {
+      refine(inputValue);
+    }
```

Comment:
```suggestion
    } else if(!searchAsYouType) {
      refine(inputValue);
    }
```
I guess it could make sense to follow [InstantSearch.js' implementation](https://github.com/algolia/instantsearch.js/blob/master/src/components/SearchBox/SearchBox.tsx#L114-L116)

Reply:
Makes sense ! I updated this and the test name accordingly 😄

Reply:
Yes you're right! I think we should call the `onSubmit` prop if it exists and only call `refine` if `searchAsYouType` is `false`, like what we're doing on InstantSearch.js (decoupled conditions).
This way, `refine` still gets called even if the user passes the `onSubmit` prop.

Reply:
Alright, I'm just running it after `refine` so that they can later refine with a custom value if they want :)